src/compiler/utilities.ts

```diff
@@ -2434,10 +2434,7 @@ namespace ts {
     }
 
     export function isIntrinsicJsxName(name: __String | string) {
-        // An escaped identifier had a leading underscore prior to being escaped, which would return true
-        // The escape adds an extra underscore which does not change the result
-        const ch = (name as string).substr(0, 1);
-        return ch.toLowerCase() === ch;

```

Comment:
`ch.toLowerCase() === ch || ch === "_"` ?

Reply:
aah.. that is the opposite.. sorry..

Reply:
```ts
const ch = (name as string).charCodeAt(0);
return (ch >= CharacterCodes.a && ch <= CharacterCodes.z) ||
 ch == CharacterCodes._ ||
 (name as string).indexof("-") > 0;
```

Reply:
and avoid the creation of a regexp on every invocation.

Reply:
@mhegazy good catch, thanks!

Reply:
oops, missed the comment before. Updating…